dascore/proc/whiten.py

```diff
@@ -3,18 +3,144 @@
 from __future__ import annotations
 
 import math
+from collections.abc import Sequence
 
 import numpy as np
-import numpy.fft as nft
 import pandas as pd
 from scipy.ndimage import convolve1d
 from scipy.signal.windows import tukey
 
 from dascore.constants import PatchType
 from dascore.exceptions import ParameterError
-from dascore.transform.fourier import _get_dft_attrs, _get_dft_new_coords
-from dascore.utils.misc import check_filter_kwargs, check_filter_range
-from dascore.utils.patch import _get_dx_or_spacing_and_axes, get_dim_sampling_rate
+from dascore.utils.misc import (
+    broadcast_for_index,
+    check_filter_range,
+    check_filter_sequence,
+)
+from dascore.utils.time import to_float
+from dascore.utils.transformatter import FourierTransformatter
+
+
+def _check_whiten_inputs(patch, smooth_size, tukey_alpha, dim, freq_range):
+    """Ensure inputs to whiten function are ok."""
+    coord = patch.get_coord(dim, require_evenly_sampled=True)
+    step = to_float(coord.step)
+    nyquist = 1 / (2 * step)
+    fft_step = nyquist / len(coord)
+
+    if tukey_alpha < 0 or tukey_alpha > 1:
+        msg = "Tukey alpha needs to be between 0 and 1"
+        raise ParameterError(msg)
+
+    if freq_range is None:
+        return nyquist
+
+    check_filter_sequence(freq_range)
+    range_min, range_max = freq_range
+
+    low = None if pd.isnull(range_min) else range_min / nyquist
+    high = None if pd.isnull(range_max) else range_max / nyquist
+    check_filter_range(nyquist, low, high, range_min, range_max)
+
+    if smooth_size is None:
+        smooth_size = range_max - range_min
+    elif smooth_size <= 0:
+        msg = "Frequency smoothing size must be positive"
+        raise ParameterError(msg)
+    elif smooth_size >= nyquist:
+        msg = "Frequency smoothing size is larger than Nyquist"
+        raise ParameterError(msg)
+
+    if ((range_max - range_min) / fft_step) < 2:
+        msg = "Frequency range is too narrow"
+        raise ParameterError(msg)
+
+    # We need at least 5 smoothing points. Before, we could increase fft resolution
```

Comment:
If you could share a reference on this, it would be greatly appreciated!

Reply:
This was how I interpreted the original implementation. Perhaps we don't need it anymore? See the following comment.